### DISCUSSION TITLE
Operator releases

### DISCUSSION THREAD

**Discussion by vdemeester** [tektoncd/operator#350] [Ideas]
I think we need to rethink the way we do Operator releases. At the time of the creation of this discussion, the latest release is [`v0.23.0-2`](https://github.com/tektoncd/operator/releases/tag/v0.23.0-2), while `tektoncd/pipeline` and other component release about 2 releases.

First, we need to **release more often**, at least as the same cadence as `tektoncd/pipeline`, which is *monthly*. We have to setup some habits around this. Have some milestone, and have some goals of what we want to ship in each new releases (operator's feature, in addition to the version of the component we ship).

We need to define the "release" triggers, aka what makes the operator do a release:
- Cadence based ? (monthly, with what is there at that point)
- Feature based ? (Adding a component schedules a release ? adding a new feature ?)
- Component based ? (new release on each component release )

In addition, we may want to rethink the versioning number
- Do we need to follow pipeline's release number ? or do we go with our own ?
- if not we need to decide when we bump X, Y and Z (in version X.Y.Z)
  + At least Y on new component release
  + Z for bugfixes
  + X on new operator feature or big upgrade of a component, or inclusion of a new
    component

Finally, we need to release all our target at the same time. Right now, we only release the `kubernetes` target, and I think this doesn't make much sense (mainly because there is a disconnect between the platforms on what they ship).

Creating this discussion to gather some ideas and feedback before trying to come up with a more *definite* plan.

**Comment by vdemeester** [comment]
cc @tektoncd/operator-collaborators @tektoncd/operator-maintainers

**Comment by nikhil-thomas** [comment]
@vdemeester Thank you for starting this discussion. :technologist: :+1: 
It is high time that we cultivate and maitain release habits/routines for tektoncd/operator

> We need to define the "release" triggers, aka what makes the operator do a release:

I think a `Cadence based ? (monthly, with what is there at that point)` will be a good start. As this will be the simpletes of all in terms of determining which versions of components got in. May be down the road we can consider a `Component based ? (new release on each component release )`. 

> Do we need to follow pipeline's release number ? or do we go with our own ?

Let us stop following pipeline's version and have our own version series. One year back following pipeline version made sense as there were only pipeline and triggers and the same reconciler installed everything. But now as there are more components, different platform support, and multiple components pipeline version based release numbers for operator are confusing.

> if not we need to decide when we bump X, Y and Z (in version X.Y.Z)
At least Y on new component release
Z for bugfixes
X on new operator feature or big upgrade of a component, or inclusion of a new
component

:+1: 

> Finally, we need to release all our target at the same time. Right now, we only release the kubernetes target, and I think this doesn't make much sense (mainly because there is a disconnect between the platforms on what they ship).

:+1:

**Comment by nikhil-thomas** [comment]
https://github.com/tektoncd/operator/blob/main/docs/release/README.md
